gallery_dl/extractor/fanbox.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.fanbox.cc/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class FanboxExtractor(Extractor):
+    """Base class for fanbox extractors"""
+    category = "fanbox"
+    root = "https://www.fanbox.cc"
+    directory_fmt = ("{category}", "{creator_id}")
+    filename_fmt = "{id}_{num}.{extension}"
+    archive_fmt = "{id}_{num}"
+    _warning = True
+
+    def items(self):
+        yield Message.Version, 1
+
+        if self._warning:
+            if "FANBOXSESSID" not in self.session.cookies:
+                self.log.warning("no 'FANBOXSESSID' cookie set")
+            FanboxExtractor._warning = False
+
+        for url, data in self.posts():
+            yield Message.Directory, data
+            yield Message.Url, url, data
+
+    def posts(self):
+        """Return all relevant post objects"""
+
+    def _pagination(self, url):
+        headers = {"Origin": self.root}
+
+        while url:
+            url = text.ensure_http_scheme(url)
+            body = self.request(url, headers=headers).json()["body"]
+            for item in body["items"]:
+                for url, data in self._get_post_data_and_urls(item["id"]):
+                    yield url, data
+
+            url = body["nextUrl"]
+
+    def _get_post_data_and_urls(self, post_id):
+        """Fetch and process post data"""
+        headers = {"Origin": self.root}
+        url = "https://api.fanbox.cc/post.info?postId="+post_id
+        pbody = self.request(url, headers=headers).json()["body"]
+
+        post = {
+            "id": pbody["id"],
+            "title": pbody["title"],
+            "fee_required": pbody["feeRequired"],
+            "published": pbody["publishedDatetime"],
+            "updated": pbody["updatedDatetime"],
+            "type": pbody["type"],
+            "tags": pbody["tags"],
+            "creator_id": pbody["creatorId"],
+            "has_adult_content": pbody["hasAdultContent"],
+            "post_url": self.root+"/@"+pbody["creatorId"]+"/posts/"+post_id,
+            "creator_user_id": pbody["user"]["userId"],
+            "creator_name": pbody["user"]["name"],
+            "text": pbody["body"].get("text", None) if pbody["body"] else None,
+            "is_cover_image": False
+        }
+
+        num = 0
+        if "coverImageUrl" in pbody and pbody["coverImageUrl"]:
+            final_post = dict(post)
+            final_post["isCoverImage"] = True
+            final_post["file_url"] = pbody["coverImageUrl"]
+            final_post = text.nameext_from_url(
+                pbody["coverImageUrl"], final_post
+            )
+            final_post["num"] = num
+            num += 1
+            yield pbody["coverImageUrl"], final_post
+
+        for group in ["images", "imageMap"]:
```

Comment:
```suggestion
        for group in ("images", "imageMap"):
```
That's just a personal pet peeve, but I don't like using lists when tuples work as well.

Reply:
Tuples take less time to create, and when you ain't modifying the list there just ain't any point in using lists instead of tuples.

So it's more than just a personal pet peeve, it's actually a better way.

Reply:
You are correct, but the difference between the two is negligible (<10%) and is therefore not really important. The difference in runtime between `dict(post)` and `post.copy()`, for example, is a lot more significant in comparison (~50%), but even that doesn't matter when compared to anything I/O related.